site/docs/components/password_input.mdx

```diff
@@ -0,0 +1,256 @@
+import { StatusLabel, PasswordInput, Button, IconEye, IconEyeCrossed } from "hds-react";
+
+import ColorBox from "../../src/components/ColorBox";
+import LargeParagraph from "../../src/components/LargeParagraph";
+import Link from "../../src/components/Link";
+import Text from "../../src/components/Text";
+
+# Password input
+
+<StatusLabel type="alert">Draft</StatusLabel>
+<StatusLabel type="success" style={{marginLeft: 'var(--spacing-xs)'}}>Accessible</StatusLabel>
+
+<LargeParagraph>
+    A password field is a foundational input field component that the user can interact with and specifically input password data.
+</LargeParagraph>
+
+## Principles
+- A label should always be provided with password fields.
+- If possible, allow the password input to reflect the strength of the password entered.
+- If possible, add other programmatic assistance. Detect and suggest changes to provide the user with assistance in generating strong passwords.
+- If possible, do input validation in client-side real-time and provide the user with immediate feedback. See [HDS Form validation pattern page](/patterns/form-validation) for more information and best practices with password fields.
+- Password fields should always be mandatory and it is recommended to mark it as a required field.
+- Use `autocomplete="new-password"` and `id="new-password"` for the password input in a sign-up form, and for the new password in a reset-password form.
+- HDS password input provides a way to mask/unmask the password. Do not disabled this functionality without a good reason.
+- Use `aria-label` and `aria-describedby` for password inputs.
+
+## Accessibility
+- In the case of an insufficient password, aim to provide clear instructions to the user on how to correct the mistake. Always provide a text description of the error.
+- It is required to use `autocomplete` attribute if input control is listed in the common input purposes list. You can read the full list of common input purposes at <Link href="https://www.w3.org/TR/WCAG21/#input-purposes" external>WCAG 2.1 documentation website</Link>. You can read more about the requirement from <Link href="https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose" external>WCAG 2.1 Identify Input Purpose guideline</Link>.
+- Place sign-in in its own `form` element and use `landmark` region to mark the login area. This allows assistive technologies to quickly locate and skip to the login section.
+
+## Security
+As HDS password input provides only the front-end solution, remember that you need to take care of the security in your project.
+Here are some good practices you need to consider.
+
+### General Security
+- Do not prevent the user from copying and pasting the password into the field
+- Prefer using ready authentication libraries and services, and familiarize yourself properly with their implementation
+- In City of Helsinki Services, always use Helsinki Profiili authentication if possible.
```

Comment:
Maybe add: "In practice, this will often mean that the service itself will not need to use the password field or any login form of its own, because account logins will forward to the Helsinki Profiili login screen."